Tests/OpenCombineTests/ObservableObjectTests.swift

```diff
@@ -0,0 +1,423 @@
+            receiveSubscription: { downstreamSubscription1 = $0 }
+        )
+
+        testObject.objectWillChange.subscribe(tracking1)
+        tracking1.assertHistoryEqual([.subscription("ObservableObjectPublisher")])
+        downstreamSubscription1?.request(.max(2))
+        tracking1.assertHistoryEqual([.subscription("ObservableObjectPublisher")])
+        testObject.state1 += 1
+        testObject.state1 += 2
+        testObject.state1 += 3
+        tracking1.assertHistoryEqual([.subscription("ObservableObjectPublisher"),
+                                      .signal,
+                                      .signal,
+                                      .signal])
+        testObject.state2 += 1
+        tracking1.assertHistoryEqual([.subscription("ObservableObjectPublisher"),
+                                      .signal,
+                                      .signal,
+                                      .signal,
+                                      .signal])
+        downstreamSubscription1?.request(.max(10))
+        tracking1.assertHistoryEqual([.subscription("ObservableObjectPublisher"),
+                                      .signal,
+                                      .signal,
+                                      .signal,
+                                      .signal])
+
+        let tracking2 = TrackingSubscriberBase<Void, Never>(
+            receiveSubscription: { $0.request(.unlimited) }
+        )
+        testObject.objectWillChange.subscribe(tracking2)
+        tracking2.assertHistoryEqual([.subscription("ObservableObjectPublisher")])
+
+        testObject.state1 = 42
+        tracking1.assertHistoryEqual([.subscription("ObservableObjectPublisher"),
+                                      .signal,
+                                      .signal,
+                                      .signal,
+                                      .signal,
+                                      .signal])
+        tracking2.assertHistoryEqual([.subscription("ObservableObjectPublisher"),
+                                      .signal])
+
+        downstreamSubscription1?.cancel()
+        testObject.state1 = -1
+
+        tracking1.assertHistoryEqual([.subscription("ObservableObjectPublisher"),
+                                      .signal,
+                                      .signal,
+                                      .signal,
+                                      .signal,
+                                      .signal])
+        tracking2.assertHistoryEqual([.subscription("ObservableObjectPublisher"),
+                                      .value(()),
+                                      .value(())])
+    }
+
+    func testNoFields() {
+        let observableObject = NoFields()
+        _ = observableObject.objectWillChange
+    }
+
+    func testNoPublishedFields() {
+        let observableObject = NoPublishedFields()
+        _ = observableObject.objectWillChange
+    }
```

Comment:
Please add a TODO comment to these tests, so that in the future, when we improve the implementation, we could verify that the returned `ObservableObjectPublisher` is the same every time